### ISSUE TITLE
Test Dataset and Validation set are same?

### ISSUE THREAD

**Issue by tangores** [lisiqi19971013/EventStream-SR#7]
Hi, I was curious to understanding whether the results presented in the paper are based on the utilization of the same test_dataset for both validation and testing within the current implementation. Your insights into this aspect would be greatly appreciated.  (I am new to spiking networks, please help me understand this aspect)

**Comment by lisiqi19971013** [comment]
Yes, we used the original data splitting of these datasets since we have to validate the effectiveness of the downstream applications.

**Comment by lisiqi19971013** [comment]
I'm not entirely sure, but I remember that none of these datasets had a validation set in their original division.

**Comment by tangores** [comment]
In the field of machine learning, it's a common practice to partition the dataset into three main parts: Training, Validation (which is used for learning the model) and the Test dataset (consisting of new, unseen data). The results from the Test dataset are then showcased to highlight how well the algorithm performs.

However, it's worth noting that using only the Test dataset for both validation and testing might not provide a comprehensive assessment of the proposed algorithm's effectiveness. A more balanced approach, involving a dedicated Validation set, is generally recommended to ensure a more accurate evaluation. This approach helps ensure that the algorithm's performance is thoroughly justified and reliable.

**Comment by lisiqi19971013** [comment]
I actually knew that, but I don't want to change the original divisions provided by the proposers of these datasets so that downstream application performance could be easily validated using existing classification models or image reconstruction models.

**Comment by tangores** [comment]
You could use the same training set and create a random subset of validation during training. Later take the unseen test dataset for further evaluation. In this way you'll not be changing the original divisions too.
